packages/messaging/src/interfaces/token-details.ts

```diff
@@ -14,8 +14,16 @@
  * limitations under the License.
  */
 
-export enum NotificationPermission {
-  DEFAULT = 'default',
-  GRANTED = 'granted',
-  DENIED = 'denied'
+export interface TokenDetails {
+  fcmToken: string;
+  swScope: string;
+  vapidKey: string | Uint8Array;
+  subscription: PushSubscription;
+  fcmSenderId: string;
+  fcmPushSet: string;
+
+  createTime?: number;
```

Comment:
Is ? still needed?

Reply:
Yep, it's not defined everywhere TokenDetails is used.

I think we should eventually split the database version and code version (which doesn't define these optional members until db operations) of TokenDetails, and make the DB version extend the regular version. WDYT?

Reply:
synced offline, leaning towrads using Partial, or just setting the pushSubscription details when the TokenDetails are created, rather than in TokenDetails model.